### PR TITLE
rex.ctl improvements

### PR DESCRIPTION
**Pull request by xitology**
Various improvements for `rex.ctl`:
- load no more than one config file
- look for the config file in the current directory, then in the virtualenv root
- parse `rex.yaml` with `rex.core.ValidatingLoader`
- derive the config file name from the executable name
- override the executable name from command-line using `@instance` notation

### CODE REVIEW

**Comment by jayclassless** [comment]
So, now these invocations would be equivalent?
- `bin/rex --config=my_application.yaml some-task`
- `bin/rex @my_application some-task`
- `bin/my_application some-task`

And, it would favor the `--config` option over `@my_application`, and favor `@my_application` over the name of the script?

**Comment by xitology** [comment]
> So, now these invocations would be equivalent?
> 
>     * `bin/rex --config=my_application.yaml some-task`
> 
>     * `bin/rex @my_application some-task`
> 
>     * `bin/my_application some-task`
> 
> 
> And, it would favor the `--config` option over `@my_application`, and favor `@my_application` over the name of the script?

Yes, that's correct

**Review by jayclassless — APPROVED** [review]

File: src/rex.ctl/README.rst [hunk 1]
```diff
@@ -105,14 +105,17 @@ If you wish to name the configuration file differently, or the file is not in
 the current directory, you can use ``--config`` parameter to specify the
 location of the configuration file::
 
-    $ rex deploy --config=/path/to/rex.yaml
+    $ rex deploy --config=myapp.yaml
     ...
 
-    $ rex serve --config=/path/to/rex.yaml
+Alternatively, you can rename the executable to match the name of the
```
Comment (andreypopp) [inline src/rex.ctl/README.rst:111]: > you can rename the executable

nit: should we have an example on how to do that in docs? Or are you planning to address that in subsequent PRs to make codebase automatically generate wrappers? (I think it's a good idea as they will show up in shell autocomplete).

File: src/rex.ctl/src/rex/ctl/core.py [hunk 2]
```diff
@@ -576,16 +577,13 @@ def prompt(msg):
...
                           description="""Command-line administration utility"""
                                       """ for the RexDB platform""",
                           local_package='rex.local',
                           entry_point='rex.ctl',
-                          config_name='rex.yaml',
-                          config_dirs=['/etc',
-                                       os.path.join(sys.prefix, '/etc'),
-                                       os.path.expanduser('~/.rex'),
-                                       os.path.abspath('.')]),
+                          config_dirs=[os.path.abspath('.'), sys.prefix]),
```
Comment (andreypopp) [inline src/rex.ctl/src/rex/ctl/core.py:585]: Do we actually need to look for configs in `.`, is it used somewhere? If not (or it's used just in a few places) I think it's better to leave just `sys.prefix` and convert those places to be explicit about their behaviour with `--config ./APP.yaml`.

**Review by andreypopp — APPROVED** [review]
Really like these changes, does using `ValidatingLoader` mean we can start using `!include` across app configs? Should we have an example in docs for that?

**Pull request merged**